www_src/lib/cartesian.js

```diff
@@ -0,0 +1,129 @@
+class Cartesian {
+
+  constructor(options) {
+    options = options || {};
+    if (!options.height || !options.width || !options.allCoords) {
+      throw new Error('You must configure height, width, allCoords');
+    }
+    this.width = options.width;
+    this.height = options.height;
+    this.gutter = options.gutter || 0;
+    this.allCoords = options.allCoords;
+  }
+
+  get edges() {
+    return this.getEdges();
+  }
+
+  // Includes edges.
+  get all() {
+    return this.allCoords.concat(this.edges);
+  }
+
+  get allX() {
+    return this.all.map(xy => xy.x);
+  }
+  get allY() {
+    return this.all.map(xy => xy.y);
+  }
+  get widthPadded() {
+    return this.width + this.gutter;
+  }
+  get heightPadded() {
+    return this.height + this.gutter;
+  }
+
+  // Calculate absolute distance from 0 to min/max
+  getAbsDistance(xOrY, maxOrMin) {
```

Comment:
Sorry it's a bit confusing.

For the bounding box, in order to calculate how much padding to put on either side of the 0 case, I need the absolute distance to the lowest value (padding-right, padding-bottom) and the absolute distance to the highest value, for both the set of x and the set of y.

saves me having to do
```
 Math.abs(Math.min(allXValues));
 Math.abs(Math.max(allXValues));
 Math.abs(Math.min(allYValues));
 Math.abs(Math.max(allYValues));
```

(the apply is just so i can turn an array of values into arguments)